factory/src/contract.rs

```diff
@@ -0,0 +1,850 @@
+/// * `count` - the count for the counter template
+/// * `description` - optional free-form text string owner may have used to describe the offspring
+#[allow(clippy::too_many_arguments)]
+fn try_create_offspring<S: Storage, A: Api, Q: Querier>(
+    deps: &mut Extern<S, A, Q>,
+    env: Env,
+    label: String,
+    entropy: String,
+    owner: HumanAddr,
+    count: i32,
+    description: Option<String>,
+) -> HandleResult {
+    let mut config: Config = load(&deps.storage, CONFIG_KEY)?;
+    if config.stopped {
+        return Err(StdError::generic_err(
+            "The factory has been stopped. No new offspring can be created",
+        ));
+    }
+
+    let factory = ContractInfo {
+        code_hash: env.clone().contract_code_hash,
+        address: env.clone().contract.address,
+    };
+
+    // generate and save new prng, and password. (we only register an offspring retuning the matching password)
+    let prng_seed: Vec<u8> = load(&deps.storage, PRNG_SEED_KEY)?;
+    let new_prng_bytes = new_entropy(&env, prng_seed.as_ref(), entropy.as_bytes());
+    save(&mut deps.storage, PRNG_SEED_KEY, &new_prng_bytes.to_vec())?;
+
+    // store the password for future authentication
+    let password = sha_256(&new_prng_bytes);
+    save(&mut deps.storage, PENDING_KEY, &password)?;
+
+    let initmsg = OffspringInitMsg {
+        factory,
+        index: config.index,
+        label: label.clone(),
+        password: password.clone(),
+        owner,
+        count,
+        description,
+    };
+    // increment the index for the next offspring
+    config.index += 1;
+    save(&mut deps.storage, CONFIG_KEY, &config)?;
+
+    let cosmosmsg = initmsg.to_cosmos_msg(
+        label,
+        config.version.code_id,
+        config.version.code_hash,
+        None,
+    )?;
+
+    Ok(HandleResponse {
+        messages: vec![cosmosmsg],
+        log: vec![],
+        data: Some(to_binary(&HandleAnswer::Status {
+            status: Success,
+            message: None,
+        })?),
+    })
+}
+
+/// Returns HandleResult
+///
+/// Registers the calling offspring by saving its info and adding it to the appropriate lists
+///
+/// # Arguments
+///
+/// * `deps` - mutable reference to Extern containing all the contract's external dependencies
+/// * `env` - Env of contract's environment
+/// * `owner` - reference to the address of the offspring's owner
+/// * `reg_offspring` - reference to RegisterOffspringInfo of the offspring that is trying to register
+fn try_register_offspring<S: Storage, A: Api, Q: Querier>(
+    deps: &mut Extern<S, A, Q>,
+    env: Env,
+    owner: HumanAddr,
+    reg_offspring: &RegisterOffspringInfo,
+) -> HandleResult {
+    // verify this is the offspring we are waiting for
+    let load_password: Option<[u8; 32]> = may_load(&deps.storage, PENDING_KEY)?;
+    let auth_password = load_password
+        .ok_or_else(|| StdError::generic_err("Unable to authenticate registration."))?;
```

Comment:
This might be a good use case for security through obscurity. We could consider making the messages the same so that the caller does not know if it was a wrong password or an error. Similar to why you wouldn't tell a caller in web2 that it was an incorrect email or password. Thoughts?

Reply:
1. so there must be a misunderstanding in your comment above. You cannot deactivate an offspring from the factory contract. You can only deactivate an offspring by sending a message to the offspring contract and then that offspring calls the factory's DeactivateOffspring HandleMsg. A user cannot call DeactivateOffspring in the factory.

Reply:
2. The password is so random that I don't think this is a vulnerability. What error do you think is appropriate here? Just no text generic error?

Reply:
The "password" bit isn't really a traditional password, its just a storage key. Every time a new contract is made, a unique 256 bit key is made for it, the child contract calls the factory with the password to prove it was made by the factory, and then it's address is registered with the factory, and the password is made invalid for future uses. Essentially, any attacker would have 1 block to go through 10^85 possible combinations everytime a new contract is generated if they want to break it

Reply:
3. regarding "the list of valid contracts could be cleared when the function to change the offspring contract's info is called, or maybe, at the very least, it should be an option". I don't think so. Because we never clear lists ever, an offspring can be moved to the inactive list by interacting with the offspring. And even if a new contract version is uploaded, the old contracts will always be supported.

It is important to remember that factory cannot deactivate offsprings in this model. Only an offspring can deactivate itself.